### ISSUE TITLE
[Content Patcher] Paint Masks not updated unless restarting the game

### ISSUE THREAD

**Issue by Leroymilo** [Pathoschild/LookupAnything#862]
When patching a paint mask, the changes are not updated according to the `Update` field in the patch but upon restarting the game. For some reason, the Farm house is a special case which is somehow more broken: the paint seem to partially disappear when toggling the patch (see screenshots). There's nothing about the relevant content patch in [the log](https://smapi.io/log/20e2e6508b8d4fb687d0e7d04002d640) other than saying it was loaded (the Steam API issue is caused by me starting the game from a terminal because the console does not open when starting from Steam for some reason).
OS : Linux Fedora

Steps to reproduce :
1) Install a content patch patching a paintable building and its paint mask (I used [my own mod](https://www.nexusmods.com/stardewvalley/mods/9704)).
2) Load a game with a paintable building on the farm.
![Buildings before being painted](https://github.com/Pathoschild/StardewMods/assets/88249282/c8702178-ac88-4454-8683-41f5dddf20c2)
3) Paint the building.
![Buildings after being painted](https://github.com/Pathoschild/StardewMods/assets/88249282/cb5d6dda-6407-44f5-b1c6-85fcf01525fc)
4) Toggle the patch (I used [GMCM](https://www.nexusmods.com/stardewvalley/mods/5098) to do this step)
![Painted buildings after enabling the patch](https://github.com/Pathoschild/StardewMods/assets/88249282/bac85a36-e2bd-4245-9763-c96b6c7e5533)
This does not change upon sleeping, even after exiting to title.
5) Sleep (to save), close the game, start the game and load the save with the painted buildings.
![Painted buildings after sleeping (to save) and restarting the game](https://github.com/Pathoschild/StardewMods/assets/88249282/491885f9-09df-4023-a132-8d4132c122fc)

Here are the changes for the stable and the farm house :
```json
		{
			"Action": "Load",
			"Target": "Buildings/Stable, Buildings/Stable_PaintMask",
			"FromFile": "assets/Buildings/Stable/{{TargetWithoutPath}}_{{season}}.png",
			"Update": "OnTimeChange"
		}
```
```json
		{
			"Action": "Load",
			"Target": "Buildings/houses, Buildings/houses_PaintMask",
			"FromFile": "assets/Buildings/Houses/{{TargetWithoutPath}}_{{season}}.png",
			"Update": "OnTimeChange"
		}
```

Changing `Action` to `EditImage` instead of `Load` did not fix the issue.

**Comment by Pathoschild** [comment]
`OnTimeChange` does nothing if the patch values didn't change since the last update, and config tokens are considered immutable so they don't count currently.

The paint mask updated fine for me with this patch, which toggles it on each time change:
```json
{
    "Action": "EditImage",
    "Target": "Buildings/houses, Buildings/houses_PaintMask",
    "FromFile": "assets/Buildings/Houses/{{TargetWithoutPath}}_{{season}}.png",
    "When": {
        "Query: {{Time}} % 20 = 0": true
    },
    "Update": "OnTimeChange"
}
```

The issue in this case is that editing config in-game doesn't correctly update affected patches. That's a bit tricky to untangle, but the upcoming Content Patcher 2.1.0 will force-update all patches when you edit config in-game to avoid the issue.

Thanks for reporting it!

**Comment by Pathoschild** [comment]
Fixed in the upcoming SMAPI 3.0.9 (which fixes paint mask changes not always propagating correctly) and Content Patcher 2.1.0 (which now force-updates all patches when you edit config in-game). Thanks for reporting it!
